l2l/optimizers/kalmanfilter/optimizer.py

```diff
@@ -0,0 +1,267 @@
+import logging
+import numpy as np
+
+from collections import namedtuple
+from l2l import get_grouped_dict
+from l2l.utils.tools import cartesian_product
+from .enkf import EnsembleKalmanFilter as EnKF
```

Comment:
I usually prefer absolute imports rather than relative.

Reply:
You mean the `from .enk ...f` ? Probably I missed that out and will change it.

Reply:
More like `from l2l.optimizers.kalmanfilter.enkf ...` since it's a lot more explicit.